app.go

```diff
@@ -0,0 +1,187 @@
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+)
+
+type Flags struct {
+	countStrings   bool
+	showDuplicates bool
+	haveRegister   bool
+	showUnique     bool
+	countSymbols   int
+	countRow       int
+}
+
+type Data struct {
+	command    *Flags
+	inputFile  io.Reader
+	outputFile io.Writer
+}
+
+func Uniq(data *Data) error {
+	var err error = nil
+	var prevValue string
+	originPrevValue := ""
+
+	lenghtStrings := 0
+	lastCount := 0
+
+	haveRows := (*data.command).countRow
+	haveSymbols := (*data.command).countSymbols
+	haveRegister := (*data.command).haveRegister
+
+	scanner := bufio.NewScanner((*data).inputFile)
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		if lenghtStrings > 0 {
+			if haveSymbols < len(originPrevValue) {
+				prevValue = originPrevValue[haveSymbols:]
+			}
+		}
+		if haveSymbols > 0 && haveSymbols < len(line) {
+			line = line[haveSymbols:]
+		}
+		if haveRegister {
+			line = strings.ToLower(line)
+			prevValue = strings.ToLower(prevValue)
+		}
+		if haveRows > 0 {
+			words := strings.Split(line, " ")
+			if haveRows < len(words) {
+				line = strings.Join(words[haveRows:], " ")
+			}
+
+			words = strings.Split(prevValue, " ")
+			if haveRows < len(words) {
+				prevValue = strings.Join(words[haveRows:], " ")
+			}
+		}
+
+		if lenghtStrings == 0 {
+			originPrevValue = scanner.Text()
+			lenghtStrings += 1
+			lastCount = 1
+		} else if prevValue != line {
+			if err := Writer(data, originPrevValue, lastCount); err != nil {
+				return err
+			}
+
+			originPrevValue = scanner.Text()
+			lastCount = 1
+		} else if prevValue == line {
+			lenghtStrings += 1
+			lastCount += 1
+		}
+	}
+
+	if err := Writer(data, originPrevValue, lastCount); err != nil {
+		return err
+	}
+
+	return err
+}
+
+func Writer(data *Data, line string, count int) error {
+	var err error = nil
+	uniqFlag := (*data.command).showUnique
+	сountFlag := (*data.command).countStrings
+	duplicateFlag := (*data.command).showDuplicates
+
+	if !сountFlag && !duplicateFlag && !uniqFlag {
+		io.WriteString((*data).outputFile, fmt.Sprintf("%s\n", line))
```

Comment:
Через Fprintln/Fprintf лаконичней гораздо

```suggestion
		fmt.Fprintln((*data).outputFile, line))
```

В остальных местах аналогично, плиз